aea/aea_builder.py

```diff
@@ -765,10 +765,11 @@ def _build_identity_from_wallet(self, wallet: Wallet) -> Identity:
         :param wallet: the wallet
         :return: the identity
         """
-        assert self._name is not None, "You must set the name of the agent."
+        if self._name is None:
```

Comment:
Note; in cases where we check for None type, we cannot use `enforce`. Mypy does not recognise it implements the check.